dynamic_instrumentation/src/instrument.rs

```diff
@@ -485,29 +527,29 @@ impl<'tcx> Visitor<'tcx> for CollectInstrumentationPoints<'_, 'tcx> {
                     .dest(&dest)
                     .add_to(self);
             }
-            Rvalue::AddressOf(_, p) => {
-                // Instrument which local's address is taken
-                self.loc(location, location.successor_within_block(), addr_local_fn)
-                    .arg_var(dest)
-                    .arg_index_of(p.local)
-                    .source(p)
-                    .dest(&dest)
-                    .instrumentation_priority(InstrumentationPriority::Early)
-                    .add_to(self);
-            }
             Rvalue::Use(Operand::Copy(p) | Operand::Move(p)) if p.is_indirect() => {
                 // We're dereferencing something, the result of which is a reference or pointer
                 self.loc(location, location.successor_within_block(), load_value_fn)
                     .arg_var(dest)
                     .dest(&dest)
                     .add_to(self);
             }
-            Rvalue::Use(Operand::Constant(..)) => {
+            Rvalue::Use(op @ Operand::Constant(..)) => {
+                let const_ty = match op_ty(op).kind() {
+                    TyKind::RawPtr(TypeAndMut { ty, .. }) => *ty,
+                    TyKind::Ref(_, ty, _) => *ty,
+                    ty @ _ => unimplemented!("Unsupported constant ty {:?}", ty)
+                };
+                let layout = ctx.layout_of(ty::ParamEnv::reveal_all().and(const_ty))
+                    .expect("Failed to compute layout of constant");
+                let size = u32::try_from(layout.size.bytes())
+                    .expect("Failed to convert local size");
```

Comment:
What size does this come up with when `op` has type `&str`?  Zero?  I haven't looked into how the size field is used - is it okay to have a zero there?

Reply:
Good catch, yeah `&str` has zero size. That's not good, I need to fix it.